src/pretix/base/models/orders.py

```diff
@@ -1219,6 +1220,38 @@ def create_transactions(self, is_new=False, positions=None, fees=None, dt_now=No
         _transactions_mark_order_clean(self.pk)
         return create
 
+    def tagged_secret(self, tag, secret_length=64):
+        return salted_hmac(key_salt=tag, value=self.secret, algorithm="sha256").hexdigest()[:secret_length]
+
+    @staticmethod
+    def get_with_secret_check(code, received_secret, tag, secret_length=64, qs=None):
```

Comment:
Django style question: Would this better be a method on a custom QuerySet class so I can chain it with .filter() or .annotate() etc?

Reply:
Chaining as in `event.orders.get_with_secret_check(...).annotate(...)`? I think it wouldn't be possible to implement it that way, as we need the Order object in here for the secret check? 

The other way round, would work I think, which would just change the API slightly from `Order.get_with_secret_check(..., qs=event.orders.annotate(...))` to `event.orders.annotate(...).get_with_secret_check(...)` which seems nicer though, yes. I'll have a look into implementing that.

Reply:
could you confirm that this is in fact an okay way to specify the Manager for orders:
https://github.com/pretix/pretix/pull/4139/files#diff-a88712c33526e582606fb2e095cea07e0972a201338a955b91cf2dfc675fa3bbR318

in other places it's done in a more verbose way with a separate manager (e.g. [ItemQuerySetManager](https://github.com/pretix/pretix/blob/master/src/pretix/base/models/items.py#L289)), I think my way is equivalent with less duplicated code but I'm not sure if there is some subtlety I'm missing?

Reply:
as_manager() is certainly fine, and if it works with scopes, fine with me